### PR TITLE
Add QuoteFields parameter to ConvertTo-Csv and Export-Csv

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Add QuoteFields parameter to ConvertTo-Csv and Export-Csv. This allows to explicitly point fields to quote in output.

## PR Context

Please look #8890.

/cc @DavidBerg-MSFT

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/4122
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Csv.Tests.ps1 [hunk 1]
```diff
@@ -90,11 +90,25 @@ Describe "ConvertTo-Csv" -Tags "CI" {
...
+        { $testObject | ConvertTo-Csv -UseQuotes Always -QuoteFields "TestFieldName" } |
+            Should -Throw -ErrorId "CannotSpecifyQuoteFieldsAndUseQuotes,Microsoft.PowerShell.Commands.ConvertToCsvCommand"
+    }
+
     It "Does not support -IncludeTypeInformation and -NoTypeInformation at the same time" {
         { $testObject | ConvertTo-Csv -IncludeTypeInformation -NoTypeInformation } |
             Should -Throw -ErrorId "CannotSpecifyIncludeTypeInformationAndNoTypeInformation,Microsoft.PowerShell.Commands.ConvertToCsvCommand"
     }
 
+    Context "QuoteFields parameter" {
```
Comment (SteveL-MSFT) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Csv.Tests.ps1]: Seems like we should test multiple fields and also case insensitivity

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Csv.Tests.ps1 (hunk 1 above)
Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Csv.Tests.ps1]: Done.

**Review by SteveL-MSFT — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs [hunk 2]
```diff
@@ -101,6 +108,13 @@ public virtual void WriteCsvLine(string line)
         /// </summary>
         protected override void BeginProcessing()
         {
+            if (this.MyInvocation.BoundParameters.ContainsKey(nameof(QuoteFields)) && this.MyInvocation.BoundParameters.ContainsKey(nameof(UseQuotes)))
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs]: Any reason we don't put them in separate parameter sets?

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs]: Complicity. We have base class with parameters and some derived classes adding parameters too - parameter sets turn into a trick.

Comment (daxian-dbw) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/CSVCommands.cs]: Got it. Thanks

**Pull request merged**